### DISCUSSION TITLE
How do I run FastStream without the CLI? (ex.: from a django command context)

### DISCUSSION THREAD

**Discussion by gawry** [ag2ai/faststream#1470] [Q&A] [answered]
I wanted to run FastStream within a django management command to get the entire Django context (and ORM).

**Comment by Lancetnik** [comment]
Hello! We already have Django integration documentation: https://faststream.airt.ai/latest/getting-started/integrations/django/

But if you want to run faststream in a separated process from django CLI, you can figure something out arounf of `app.run()` method

Also, you can take a look at #854 discussion - it can be helpful in your case

**Reply by gawry** [reply]
So, it does not seem quite what I need. What I was trying to do is to have an out-of-process with the django context so I could run orm queries. But as I said, not within the same process. 

I haven't seen many docs around the app.run() method. Is there any reference (besides digging into the code?

**Reply by Lancetnik** [reply, marked as answer]
Just `asyncio.run(app.run())`, I don't have any advices about it

You should google smth like `setup django ORM without manage.py` - it is not realted to FastStream and I don't remember, how to do it

**Reply by gawry** [reply]
I feel a little dumb but that was the missing part. Had to add it to a running event loop. 😅

**Comment by 0legRadchenko** [comment]
> I wanted to run FastStream within a django management command to get the entire Django context (and ORM).

**That's how I managed to launch it:**

```
import logging

from django.contrib.auth import get_user_model
from django.core.management.base import BaseCommand
from faststream import FastStream

from asgi import broker

logger = logging.getLogger(__name__)
User = get_user_model()


class Command(BaseCommand):
    def handle(self, *args, **kwargs):
        app = FastStream(broker=broker)

        @broker.subscriber("some-exchange")
        async def logger(log):
            user = await User.objects.alast()
            print(user, log)

        from asyncio import run
        run(app.run())
```



![image](https://github.com/airtai/faststream/assets/32248424/a6d44bd8-a600-47cb-ab3f-0b0be9d58112)
![image](https://github.com/airtai/faststream/assets/32248424/aafeb608-56e3-4d03-817d-fa8405936d8f)
